### PR TITLE
Add AliTPCCommon library recipe, make O2 and AliRoot depend on it

### PR DESCRIPTION
**Pull request by davidrohr**

### CODE REVIEW

**Comment by davidrohr** [comment]
@dberzano : Ciao Dario, this is the first PR that adds the recipe for AliTPCCommon and makes it a dependency for AliRoot and O2. It doesn't change anything for O2 / AliRoot yet. When this is merged, I'll open the PRs for AliRoot and for O2.

I added the dependency to AliRoot in all the defaults files except the RHEL6 based ones. Hope this is correct in this way.

Cheers
David

**Comment by dberzano** [comment]
I am having a look.

File: aliroot.sh [hunk 1]
```diff
@@ -8,6 +8,7 @@ requires:
   - GEANT3
   - GEANT4_VMC
   - Vc
+  - AliTPCCommon
```
Comment (dberzano) [inline aliroot.sh]: So, as we've discussed, `AliTPCCommon` should be visible at build time only to AliRoot and O2. Hence, it should be under the `build_requires:` stanza.

File: aliroot.sh [hunk 2]
```diff
@@ -49,6 +50,7 @@ cmake $SOURCEDIR                                                     \
       ${FASTJET_ROOT:+-DFASTJET="$FASTJET_ROOT"}                     \
       ${DPMJET_ROOT:+-DDPMJET="$DPMJET_ROOT"}                        \
       ${ZEROMQ_ROOT:+-DZEROMQ=$ZEROMQ_ROOT}                          \
+      ${ALITPCCOMMON_ROOT:+-DALITPCCOMMON_DIR=$ALITPCCOMMON_ROOT}    \
```
Comment (dberzano) [inline aliroot.sh]: Here I am assuming that the CMake variable `ALITPCCOMMON_DIR` will be properly dealt with in a separate AliRoot PR, am I right?

File: alitpccommon.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package: AliTPCCommon
+version: "%(tag_basename)s"
+tag: alitpccommon-v1.0
+source: https://github.com/AliceO2Group/AliTPCCommon
+build_requires:
+  - CMake
+---
+#!/bin/bash -e
+cmake ${SOURCEDIR}/ \
```
Comment (dberzano) [inline alitpccommon.sh]: Please align the `\` properly:

```bash
cmake ${SOURCEDIR}/                        \
      -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE \
      -DCMAKE_INSTALL_PREFIX=${INSTALLROOT}
```

Thanks!

File: alitpccommon.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+mkdir -p "$MODULEDIR"
+cat > "$MODULEFILE" <<EoF
+#%Module1.0
+proc ModulesHelp { } {
+  global version
+  puts stderr "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@@"
+}
+set version $PKGVERSION-@@PKGREVISION@$PKGHASH@@
+module-whatis "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@@"
+# Dependencies
```
Comment (dberzano) [inline alitpccommon.sh]: I know this Modulefile serves no purpose (it's a build requirement after all), however can I ask you to set `ALITPCCOMMON_ROOT` only?

File: defaults-o2-prod.sh [hunk 5]
```diff
@@ -46,6 +46,7 @@ overrides:
       - GEANT4_VMC
       - Vc
       - ZeroMQ
+      - AliTPCCommon
```
Comment (dberzano) [inline defaults-o2-prod.sh]: Should be under `build_requires:`

File: defaults-o2.sh [hunk 6]
```diff
@@ -49,6 +49,7 @@ overrides:
       - GEANT4_VMC
       - Vc
       - ZeroMQ
+      - AliTPCCommon
```
Comment (dberzano) [inline defaults-o2.sh]: Should be under `build_requires:`

File: defaults-user-root6.sh [hunk 7]
```diff
@@ -12,6 +12,7 @@ overrides:
       - ROOT
       - fastjet:(?!.*ppc64)
       - Vc
+      - AliTPCCommon
```
Comment (dberzano) [inline defaults-user-root6.sh]: Should be under `build_requires:`

File: defaults-user.sh [hunk 8]
```diff
@@ -12,6 +12,7 @@ overrides:
       - ROOT
       - fastjet:(?!.*ppc64)
       - Vc
+      - AliTPCCommon
```
Comment (dberzano) [inline defaults-user.sh]: Should be under `build_requires:`

File: o2.sh [hunk 9]
```diff
@@ -12,6 +12,7 @@ requires:
   - Configuration
   - Monitoring
   - ms_gsl
+  - AliTPCCommon
```
Comment (dberzano) [inline o2.sh]: Should be under `build_requires:`

File: o2.sh [hunk 10]
```diff
@@ -118,6 +119,7 @@ cmake $SOURCEDIR -DCMAKE_INSTALL_PREFIX=$INSTALLROOT
       -DCMAKE_EXPORT_COMPILE_COMMANDS=ON                                                    \
       ${CXXSTD:+-DCMAKE_CXX_STANDARD=$CXXSTD}                                               \
       ${O2HLTCATRACKING_ROOT:+-DO2_TPCCA_TRACKING_LIB_DIR=$O2HLTCATRACKING_ROOT}            \
+      ${ALITPCCOMMON_ROOT:+-DALITPCCOMMON_DIR=$ALITPCCOMMON_ROOT}                           \
```
Comment (dberzano) [inline o2.sh]: I am again assuming that proper O2 support will be added

**Comment by davidrohr** [comment]
@dberzano : Hi Dario, of course, as you say, the ALITPCCOMMON_DIR will be used by upcoming O2 and ALIROOT pull requests. I have moved the requirement to build requirements, and also applied the other changes you suggested. Tried the full thing here including the upcoming O2 and AliRoot PRs and it works.

Cheers
David

**Pull request merged**